tools/clang/lib/SPIRV/Structure.cpp

```diff
@@ -143,6 +143,34 @@ void SPIRVModule::clear() {
   functions.clear();
 }
 
+void SPIRVModule::takeIntegerTypes(InstBuilder *ib) {
+  const auto &consumer = ib->getConsumer();
+  // If it finds any integer type, feeds it into the consumer, and removes it
+  // from the types collection.
+  types.remove_if([&consumer](std::pair<const Type *, uint32_t> &item) {
+    bool isInteger = item.first->isIntegerType();
+    if (isInteger)
+      consumer(item.first->withResultId(item.second));
+    return isInteger;
+  });
+}
+
+void SPIRVModule::takeConstantForArrayType(const Type *arrType,
+                                           InstBuilder *ib) {
+  const auto &consumer = ib->getConsumer();
+  uint32_t arrayLengthResultId = arrType->getArgs().back();
```

Comment:
const uint32_t

Maybe we should put an assert here to guard against calling this method with non-array type?